packages/core/src/RenderingEngine/VolumeViewport.ts

```diff
@@ -247,10 +247,11 @@ class VolumeViewport extends BaseVolumeViewport {
     resetZoom = true,
     resetToCenter = true,
     resetRotation = false,
-    supressEvents = false
+    supressEvents = false,
+    resetOrientation = true
   ): boolean {
     const { orientation } = this.viewportProperties;
-    if (orientation) {
+    if (orientation && resetOrientation) {
```

Comment:
This fixes the stack overflow because it avoids trying to recursively reset the stack.